core/src/main/java/brooklyn/util/config/ConfigBag.java

```diff
@@ -132,12 +166,12 @@ public String getDescription() {
     
     /** current values for all entries 
      * @return non-modifiable map of strings to object */
-    public Map<String,Object> getAllConfig() {
-        return Collections.unmodifiableMap(config);
+    public synchronized Map<String,Object> getAllConfig() {
```

Comment:
Ah, I see that `getAllConfigMutable` means that we may need to synchronize on this. I really don't like our uses of `getAllConfigMutable`. Having public methods like that is an anti-pattern in my opinion.

Longer term, we should deprecate that and also `getUnusedConfigMutable`. We could have a `getMutex()` method for what to synchronize on, with that marked `@Beta`?